drivers/sensor/sx9500/sx9500_trigger.c

```diff
@@ -129,7 +129,7 @@ static void sx9500_work_cb(struct k_work *work)
 int sx9500_setup_interrupt(const struct device *dev)
 {
 	struct sx9500_data *data = dev->data;
-	const struct device *gpio;
+	struct gpio_dt_spec int_gpio = GPIO_DT_SPEC_INST_GET(0, int_gpios);
```

Comment:
should be stored in sensor's config

Reply:
I agree with @gmarull, makes drivers consistent and allows for easy access if this driver is updated to use the gpio elsewhere in the future.

Reply:
@MaureenHelm thoughts on this?

Reply:
I agree it would be nice to do this, but I don't understand the point about easy access by @benjaminbjornsson . IMO, this would bring the driver closer to supporting multiple instances.

Reply:
the reason it should be in config is to facilitate transition to multi-instance

Reply:
Easy access as in having the gpio struct readily available in `dev->config` if needed in future updates.